gammapy/datasets/actors.py

```diff
@@ -0,0 +1,225 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import copy
+import inspect
+import logging
+import numpy as np
+from gammapy.modeling.models import DatasetModels
+from .core import Dataset, Datasets
+from .map import MapDataset
+
+log = logging.getLogger(__name__)
+
+
+__all__ = ["DatasetsActor"]
+
+
+class DatasetsActor(Datasets):
+    """A modified Dataset collection for parallel evaluation using ray actors.
+    Support only datasets composed of MapDataset.
+
+    Parameters
+    ----------
+    datasets : `Datasets`
+        Datasets
+    """
+
+    def __init__(self, datasets=None):
+        from ray import get
+
+        log.warning(
+            "Gammapy support for parallelisation with ray is still a prototype and is not fully functional."
+        )
+
+        if datasets is not None:
+            datasets_list = []
+            while datasets:
+                d0 = datasets[0]
+                if d0.tag != "MapDataset":
+                    raise TypeError(
+                        f"For now datasets parallel evaluation is only supported for MapDataset, got {d0.tag} instead"
+                    )
+                datasets_list.append(MapDatasetActor(d0))
+                datasets.remove(d0)  # moved to remote so removed from main process
+            self._datasets = datasets_list
+            self._ray_get = get
+
+        # trigger actors auto_init_wrapper (so overhead so appears on init)
+        self.name
```

Comment:
added this to ensure that the actors setup is done on init so the time is not counted in the first method used
<img width="1423" alt="Screenshot 2023-11-20 at 13 43 48" src="https://github.com/gammapy/gammapy/assets/25845654/bff8c1fc-bf6e-46a4-b22b-3aafa5ae9446">
so the method I mentioned don't add that much time, it's just the creation of new remote actors for each energy bins (that could be solved using masking instead of slicing).

Reply:
yes slicing will create new datasets/actors so the data will be copied and ray will have to start a new kernel to perform the computations on this dataset.